chrome-extension/sidebar/sidebar.js

```diff
@@ -48,6 +48,318 @@ const htmlToElement = (html) => {
+  // Looks like { "CSCI4011":26, "CSCI4041":37 ... }
+  const courseSizeDict = {};
+
+  // Get the main container with class 'course-list-results'
+  let courseListResults = document.querySelector('.course-list-results');
+  let courseDivs = Array.from(courseListResults.firstElementChild.children);
+
+  for (let courseDiv of courseDivs) {
+    // Get name of course
+    let courseName = courseDiv.querySelector('a').getAttribute('name');
+    
+    // Get seats available value
+    let targetDiv = courseDiv.querySelector('div:nth-child(2) > div:nth-child(3)');
+    let trElements = targetDiv.querySelectorAll('tr');
+    
+    let sum = 0;
+    trElements.forEach(tr => {
+        let lastChildText = tr.lastElementChild.innerText;
+        sum += parseFloat(extractDifference(lastChildText)) || 0; // Ensure the text is converted to a number
+    });
+    
+    courseSizeDict[courseName] = sum;
+  }
+
+  // sort the courseDivs in the order we want
+  courseDivs.sort((a, b) => {
+    let nameA = a.querySelector('a').getAttribute('name');
+    let nameB = b.querySelector('a').getAttribute('name');
+    return courseSizeDict[nameB] - courseSizeDict[nameA];
+  });
+
+  console.log(courseSizeDict); // For debugging purposes, display the updated dictionary
+  console.log(courseDivs); // For debugging purposes, display the updated divs
+
+  // remove all the courses to add our own ordering of the courses
+  courseListResults.firstElementChild.innerHTML = '';
+  courseDivs.forEach(courseDiv => {
+      courseListResults.firstElementChild.appendChild(courseDiv);
+  });
+
+  // click all the 'hide' buttons
+  const buttons2 = document.querySelectorAll('.action-sections.btn.btn-default.pull-right');
+  buttons2.forEach(button => button.click());
+
+  // ensure this value is selected
+  document.querySelector('.size-dropdown').value = 'B';
+};
+
+// Helper function to determine seats available
+function extractDifference(text) {
+    let matches = text.match(/(\d+)\s+of\s+(\d+)/g);
+    let totalDifference = 0;
+    
+    if (matches) {
+        matches.forEach(match => {
+            let parts = match.match(/(\d+)\s+of\s+(\d+)/);
+            let current = parseInt(parts[1]);
+            let total = parseInt(parts[2]);
+            totalDifference += (total - current);
+        });
+    }
+    return totalDifference;
+}
+
+const sortByMostCommonGrade = () => {
+  let courseMostCommonGradeDict = {};
+
+  // Get the main container with class 'course-list-results'
+  let courseListResults = document.querySelector('.course-list-results');
+  let courseDivs = Array.from(courseListResults.firstElementChild.children);
+
+  // Array to hold all fetch promises
+  let fetchPromises = [];
+
+  for (let courseDiv of courseDivs) {
+    let courseName = courseDiv.querySelector('a').getAttribute('name');
+    let fullURL = `https://corsproxy.io/?https://umn.lol/class/${courseName}?static=all`; // we need to use corsproxy.io to bypass CORS restriction
```

Comment:
Use the API instead of parsing the html on this page

ie, for CSCI2021, load 

https://umn.lol/api/class/CSCI2021

and that should also fix your CORS issues

Reply:
The corsproxy.io is a workaround for chrome extensions to call API / get anything from the web iirc. I have the same problem I had before when I tried to fetch before. Since I already built everything using the website, it wouldn't make much of a different if I use the umn.lol api now. In a future update I could try to switch for code pretty purposes but I don't think it's necessary?
![Screenshot 2024-06-20 014400](https://github.com/samyok/gophergrades/assets/72324738/1ec5211c-299b-4124-a0d4-3ff3e9a66923)

Reply:
CORS is the lack of a header on the resource you are fetching, in this case our own API. The correct thing to do here would be to change the API endpoint to allow CORS rather than using a corsproxy